src/snowflake/snowpark/mock/_analyzer.py

```diff
@@ -662,9 +662,30 @@ def do_resolve_with_resolved_children(
             )
 
         if isinstance(logical_plan, TableFunctionJoin):
-            self._conn.log_not_supported_error(
-                external_feature_name="table_function.TableFunctionJoin",
-                raise_error=NotImplementedError,
+            # standard optimizer returns here:
+            # return self.plan_builder.join_table_function(
+            #     self.analyze(
+            #         logical_plan.table_function, df_aliased_col_name_to_real_col_name
+            #     ),
+            #     resolved_children[logical_plan.children[0]],
+            #     logical_plan,
+            #     logical_plan.left_cols,
+            #     logical_plan.right_cols,
+            #     self.session.conf.get("use_constant_subquery_alias", False),
+            # )
+            # func = self.analyze(...)
+            # child = resolved_children[logical_plan.children[0]]
+            # source_plan = logical_plan
+            # left_cols = logical_plan.left_cols
+            # right_cols = logical_plan.right_cols
+            # use_constant_subquery_alias = self.session.conf.get(
+            #    "use_constant_subquery_alias", False
+            # )
```

Comment:
instead of commenting the code here, you could add it when the necessary changes are made - it would be cleaner and reduce the diff size too

Reply:
I left it here, because it is hard else to understand/debug what is support to happen here.